src/test/java/de/tum/in/www1/artemis/ArchitectureTest.java

```diff
@@ -216,4 +222,32 @@ public void check(JavaMethod item, ConditionEvents events) {
             }
         };
     }
+
+    private static final Set<String> SQL_KEYWORDS = Set.of("SELECT", "UPDATE", "SET", "DELETE", "DISTINCT", "EXISTS", "FROM", "WHERE", "LEFT", "OUTER", "INNER", "JOIN", "FETCH",
+            "TREAT", "AND", "OR", "AS", "ON", "ORDER", "BY", "ASC", "DSC", "GROUP", "COUNT", "SUM", "AVG", "MAX", "MIN", "IS", "NOT", "FALSE", "TRUE", "NULL", "LIKE", "IN",
+            "BETWEEN", "HAVING", "EMPTY", "MEMBER", "OF", "UPPER", "LOWER", "TRIM");
+
+    private ArchCondition<JavaMethod> useUpperCaseSQLStyle() {
+        return new ArchCondition<>("@Query content should follow the style guide") {
+
+            @Override
+            public void check(JavaMethod item, ConditionEvents events) {
+                var queryAnnotation = item.getAnnotations().stream().filter(simpleNameAnnotation("Query")).findAny();
+                if (queryAnnotation.isEmpty()) {
+                    return;
+                }
+                Object valueProperty = queryAnnotation.get().getExplicitlyDeclaredProperty("value");
+                if (!(valueProperty instanceof String query)) {
+                    return;
+                }
+                String[] queryWords = query.split("[\\r\\n ]+");
+
+                for (var word : queryWords) {
+                    if (SQL_KEYWORDS.contains(word.toUpperCase()) && !StringUtils.isAllUpperCase(word)) {
```

Comment:
I think as we split on every space, we get quite some empty words here (indentation). I think we should check for emptiness ferst to prevent a set query on every call.

Reply:
We spilt on "[\\r\\n ]+" so we remove all line breaks and spaces at all. There will be no empty words in the array.

Reply:
Huh? But if I have the word `    ` and would split it along ` `, I'd expect the result to be `['', '', '', '', '']`.
In other words, if I'd split `   ` along the second space, I'd expect `[' ', ' ']`

Reply:
![grafik](https://github.com/ls1intum/Artemis/assets/26540346/ca2e9ffc-c959-47d6-8ae7-f8c84cff276d)

Reply:
![](https://gyazo.com/2c4b88e16d53051e471c76f21f2da741/raw)
I love methods that have specialised behaviour for one specific input.